pyccel/parser/semantic.py

```diff
@@ -2576,12 +2569,11 @@ def _visit_FunctionDef(self, expr, **settings):
 
                 results = new_results
 
-            if len(interfaces) == 1:
-                # insert the FunctionDef into the scope
-                # to handle the case of a recursive function
-                # TODO improve in the case of an interface
-                func = FunctionDef(name, args, results, [])
-                self.insert_function(func)
+            # insert the FunctionDef into the scope
+            # to handle the case of a recursive function
+            # TODO improve in the case of an interface
```

Comment:
you can delete this TODO, can you confirm that the interface works when we have a recursive function ?

Reply:
Please create a new issue if this does not work.

Reply:
Yes I will create an issue for it

Reply:
Issue #551 is created for this